main.tf

```diff
@@ -21,7 +21,7 @@ resource "aws_vpn_gateway" "default" {
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
   count            = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
-  device_name      = module.this.id
+  device_name      = var.customer_gateway_device_name
```

Comment:
this is not backwards compatible.
please add a tenary to always default to module.this.id if var.customer_gateway_device_name is not defined.

Reply:
@jamengual - I have added the the default to be an empty string which will be converted to the module id. But to allow backwards compatibility null is a valid option